player/scripting.c

```diff
@@ -255,14 +255,24 @@ typedef int (*mpv_open_cplugin)(mpv_handle *handle);
 static int load_cplugin(struct mpv_handle *client, const char *fname)
 {
     int r = -1;
+    char *err;
+    MPContext *ctx = mp_client_get_core(client);
     void *lib = dlopen(fname, RTLD_NOW | RTLD_LOCAL);
-    if (!lib)
+    if (!lib) {
+        err = dlerror();
+        if (err)
+            MP_ERR(ctx, "C plugin library load error: '%s'\n", err);
         goto error;
+    }
     // Note: once loaded, we never unload, as unloading the libraries linked to
     //       the plugin can cause random serious problems.
     mpv_open_cplugin sym = (mpv_open_cplugin)dlsym(lib, MPV_DLOPEN_FN);
-    if (!sym)
+    if (!sym) {
+        err = dlerror();
+        if (err)
+            MP_ERR(ctx, "C plugin symbol load error: '%s'\n", err);
         goto error;
```

Comment:
Why duplicate the code instead of querying the error in the error handling block? (Would require a separate label for dlerror cases or so.)